app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -50,10 +50,13 @@ def save_inventory_multi(association, hashes, deletes, find_key, child_keys = []
     remove_keys = Array.wrap(extra_keys) + child_keys
 
     record_index = TypedIndex.new(association, find_key)
+    reconnect_index = TypedIndex.new(reconnect_from, find_key)
```

Comment:
hm, I rewrote the saving code in the graph refresh to not do Model.all, but Model.find_each. And we could also limit reconnect_from.where(:ems_ref => [:ems_refs_to_be_created])

but we can't do any of this, unless we would rewrite the old refresh saving code entirely, like in graph refresh. :-)

well, theoretically we could use new_records and the ems_refs there to limit the reconnect_from further, that should help a lot. Since I expect we will be loading just dozens of 'to be reconnected' records, comparing to possible 100k+ of disconnected records in total.

The initial refresh could also build a huge query, so might be good to do it in the batches